src/main.rs

```diff
@@ -223,6 +245,96 @@ fn tracing_panic_hook(panic_info: &PanicInfo) {
 	);
 }
 
+async  fn install_wakatime_cli_if_missing() -> Result<()> {
+	if which::which("wakatime-cli").is_ok() {
+		tracing::debug!("wakatime-cli is already installed");
+		return Ok(());
+	}
+
+	let download_url = format!("https://github.com/wakatime/wakatime-cli/releases/latest/download/wakatime-cli-{WAKATIME_CLI_RELEASE_FILE_PLATFORM}-{WAKATIME_CLI_RELEASE_FILE_ARCH}.zip");
+	tracing::info!("Downloading wakatime-cli from {download_url}");
+
+	// Create a HTTP client
+	let client = HttpClient::new();
+
+	// Send a GET request to download the file
+	let response = client.get(&download_url).send().await.map_err(|e| Error {
+		code: tower_lsp::jsonrpc::ErrorCode::InternalError,
+		data: None,
+		message: std::borrow::Cow::Owned(format!("Failed to download wakatime-cli: {e}")),
+	})?;
+
+	// Ensure the request was successful
+	if !response.status().is_success() {
+		let status_code = response.status();
+		return Err(Error {
+			code: tower_lsp::jsonrpc::ErrorCode::InternalError,
+			message: std::borrow::Cow::Owned(format!(
+				"Failed to download wakatime-cli: got HTTP {status_code} on {download_url}"
+			)),
+			data: None,
+		});
+	}
+
+	// Read the response body into a Vec<u8>
+	let bytes = response.bytes().await.map_err(|e| Error {
+		code: tower_lsp::jsonrpc::ErrorCode::InternalError,
+		data: None,
+		message: std::borrow::Cow::Owned(format!(
+			"While downloading wakatime-cli: failed to read response body: {e}"
+		)),
+	})?;
+	let reader = Cursor::new(bytes);
+
+	tracing::debug!("Extracting wakatime-cli from zip archive");
+
+	// Create a ZipArchive from the response body
+	let mut archive = ZipArchive::new(reader).map_err(|e| Error {
+		code: tower_lsp::jsonrpc::ErrorCode::InternalError,
+		data: None,
+		message: std::borrow::Cow::Owned(format!(
+			"While downloading wakatime-cli: could not parse zip archive: {e}"
+		)),
+	})?;
+
+	// Extract the first (and presumably only) file from the archive
+	let mut file = archive.by_index(0).map_err(|e| Error {
+		code: tower_lsp::jsonrpc::ErrorCode::InternalError,
+		data: None,
+		message: std::borrow::Cow::Owned(format!(
+			"While downloading wakatime-cli: archive has no files (or is corrupted): {e}"
+		)),
+	})?;
+
+	let output_filepath = env::current_exe()
+		.expect("Could not get current executable path")
+		.parent()
+		.expect("Could not get parent directory of current executable path")
+		.join("wakatime-cli");
+
+	// Create the output file, store it next to the current executable
+	let mut output_file = File::create(&output_filepath).map_err(|e| Error {
+		code: tower_lsp::jsonrpc::ErrorCode::InternalError,
+		data: None,
+		message: std::borrow::Cow::Owned(format!(
+			"While downloading wakatime-cli: could not create output file: {e}"
+		)),
+	})?;
```

Comment:
https://wakatime.com/help/creating-plugin#plugin-init:wakatime-cli:~:text=then%20unzip%20it%20into%20~/.wakatime/
woops, seems like it should be installed in ~/.wakatime instead

Reply:
Even the log output should be changed from `/tmp`. You can use this https://github.com/muesli/go-app-paths#directories as direction on how to stop program files